### PR TITLE
upgrade to SDK 3.0.100-preview3-010431

### PR DESCRIPTION
**Pull request by mashbrno**
Upgrade to SDK 3.0.100-preview3-010431

### CODE REVIEW

File: src/Blazor.Fluxor/BrowserInterop.cs [hunk 1]
```diff
@@ -22,10 +22,24 @@ public static string GetClientScripts()
...
+    if (__intervalid)
+		clearTimeout(__intervalid);
+    
+}}
+else 
+ {{
+	__intervalid = setTimeout(invokenet, 100);	
+}}
+}}
+invokenet();
```
Comment (mrpmorris) [inline src/Blazor.Fluxor/BrowserInterop.cs]: What is the purpose of the changes to this file? Can you please explain the problem it fixes, and how to reproduce the problem?

File: src/Blazor.Fluxor/DependencyInjection/DependencyScanner.cs [hunk 2]
```diff
@@ -13,16 +13,18 @@ internal static class DependencyScanner
...
-			IEnumerable<Assembly> allCandidateAssemblies = assembliesToScan.Select(x => x.Assembly)
+			List<Assembly> allCandidateAssemblies = assembliesToScan.Select(x => x.Assembly)
 				.Union(scanWhitelist.Select(x => x.Assembly))
-				.Distinct();
+				.Distinct()
+				.ToList();
+			IEnumerable<Type> allCandidateTypes = allCandidateAssemblies.SelectMany(a => a.GetTypes())
+				.Where(t => !t.IsAbstract)
+				.ToList();
+
```
Comment (mrpmorris) [inline src/Blazor.Fluxor/DependencyInjection/DependencyScanner.cs]: Could you please give details of the problem, how to reproduce it, and how your change fixes it?

File: src/Blazor.Fluxor/Feature.cs [hunk 3]
```diff
@@ -103,7 +105,11 @@ private void TriggerStateChangedCallbacks()
 					// Keep a reference to the subscribers stop them being collected before we have finished
 					subscribers.Add(subscriber);
 					// Create a callback
-					callbacks.Add(() => BlazorComponentStateHasChangedMethod.Invoke(subscriber, null));
+					Action workItem = () => { BlazorComponentStateHasChangedMethod.Invoke(subscriber, null); };
+					callbacks.Add(() => {
+						BlazorComponentInvokeMethod.Invoke(subscriber, new object[] { workItem });
+						
+						});
```
Comment (mrpmorris) [inline src/Blazor.Fluxor/Feature.cs]: Could you please explain the problem this solves, and explain how to reproduce the problem?

File: src/Blazor.Fluxor/ReduxDevTools/ReduxDevToolsInterop.cs [hunk 4]
```diff
@@ -103,10 +108,10 @@ internal static string GetClientScripts()
 			fluxorDevTools.send(action, state);
 		}};
 
-		// Notify Fluxor of the presence of the browser plugin
+		/* Notify Fluxor of the presence of the browser plugin */
 		const detectedMessage = {{
 			payload: {{
-				type: '{ReduxDevToolsInterop.FromJsDevToolsDetectedActionTypeName}'
+				type: '{FromJsDevToolsDetectedActionTypeName}'
```
Comment (mrpmorris) [inline src/Blazor.Fluxor/ReduxDevTools/ReduxDevToolsInterop.cs]: Why was the class name removed?

File: src/Blazor.Fluxor/Store.cs [hunk 5]
```diff
@@ -133,13 +133,14 @@ public RenderFragment Initialize()
 					if (middlewareScript != null)
 					{
 						renderer.OpenElement(sequence, "script");
-						renderer.AddContent(sequence, $"// Middleware scripts: {middleware.GetType().FullName}\r\n{middlewareScript}");
+						renderer.AddContent(sequence, $"/* Middleware scripts {middleware.GetType().FullName} */ ");
+						renderer.AddMarkupContent(sequence, middlewareScript);
```
Comment (mrpmorris) [inline src/Blazor.Fluxor/Store.cs]: Why AddMarkupContent instead of AddContent?

Comment (mashbrno) [reply src/Blazor.Fluxor/Store.cs]: Because AddContent is escaping characters which caused the whole script was commented by first `//`

File: src/Blazor.Fluxor/ReduxDevTools/ReduxDevToolsInterop.cs (hunk 4 above)
Comment (mashbrno) [reply src/Blazor.Fluxor/ReduxDevTools/ReduxDevToolsInterop.cs]: This change can be ignored, caused by automatic code cleanup of R#

File: src/Blazor.Fluxor/DependencyInjection/DependencyScanner.cs (hunk 2 above)
Comment (mashbrno) [reply src/Blazor.Fluxor/DependencyInjection/DependencyScanner.cs]: Becase all my Reducers derives from my abstract BaseReducer. Previous implementation was trying to add even this abstract class to ServiceCollection which was causing error.

File: src/Blazor.Fluxor/BrowserInterop.cs (hunk 1 above)
Comment (mashbrno) [reply src/Blazor.Fluxor/BrowserInterop.cs]: With preview3 the code of the page is pre-rendered to `<app>` element on the initial load. This is causing that Fluxor middleware scripts are loaded before framework's script and its `DotNet` class is loaded.

File: src/Blazor.Fluxor/Feature.cs (hunk 3 above)
Comment (mashbrno) [reply src/Blazor.Fluxor/Feature.cs]: `StateHasChanged` has to be called on UI thread, otherwise null reference exception is thrown

**Comment by mrpmorris** [comment]
I've updated Fluxor to support 0.9.0

Could you please make changes against the new master branch. Also, could you please
1: Create an example app demonstrating the problem with any bugs
2: Have only one issue per pull request - they have more chance of being pulled in that way

Thanks!

**Comment by mashbrno** [comment]
obsolete
